src/Symfony/Bundle/TwigBundle/Resources/config/twig.php

```diff
@@ -79,8 +81,24 @@
         ->set('twig.template_iterator', TemplateIterator::class)
             ->args([service('kernel'), abstract_arg('Twig paths'), param('twig.default_path'), abstract_arg('File name pattern')])
 
+        ->set('twig.template_cache.runtime_cache', FilesystemCache::class)
+            ->args([param('kernel.cache_dir').'/twig', abstract_arg('Twig filesystem cache options')])
+
+        ->set('twig.template_cache.readonly_cache', ReadOnlyFilesystemCache::class)
+            ->args([param('kernel.build_dir').'/twig'])
+
+        ->set('twig.template_cache.warmup_cache', FilesystemCache::class)
+            ->args([param('kernel.build_dir').'/twig'])
+
+        ->set('twig.template_cache.chain', ChainCache::class)
+            ->args([[service('twig.template_cache.runtime_cache'), service('twig.template_cache.readonly_cache')]])
```

Comment:
Doesn't this mean that we will do 2 filesystem checks for each cache access in production (due to having the cached template in the readonly cache that has been warmed up and comes second) ? During the SymfonyCon keynote, @fabpot was showing a code snippet putting the readonly cache first and the runtime cache second (which would be used by runtime templates that cannot be warmed up)

Reply:
It does indeed. I put them in that order so that warmed up templates can be overridden in case of changes. But maybe it is a scenario that is neither desirable nor really useful ? I don't use twig enough on a personal level to tell.

And now, what that in mind, I better understand your comment about disabling it when hot-reloading is enabled !